crates/wassette/src/runtime_context.rs

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+//! Support utilities for sharing Wasmtime engine and linker state across lifecycle
+//! manager instances.
+
+use std::sync::Arc;
+
+use anyhow::Result;
+use wasmtime::component::{Component, InstancePre, Linker};
+use wasmtime::Engine;
+use wasmtime_wasi_config::WasiConfig;
+
+use crate::{WasiState, WassetteWasiState};
+
+/// Encapsulates Wasmtime engine and linker setup for reuse across the lifecycle manager.
+#[derive(Clone)]
+pub struct RuntimeContext {
+    engine: Arc<Engine>,
+    linker: Arc<Linker<WassetteWasiState<WasiState>>>,
+}
+
+impl RuntimeContext {
```

Comment:
I love having this type, but I think it would be better if we exposed the linker and the engine via `AsRef` and `Deref` instead of exposing through our own methods here. This leaves things flexible for later as well if we need to use any other methods from the types. It also means we can just use this type as an `Engine` or `Linker` automatically